src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -1913,6 +1919,10 @@ public void close() {
         buffer = null;
         containerStack = null;
         byteBuffer = null;
+        terminate();
+        // Use a unified code path for all cursors after close. This path forces a termination check before accessing
+        // the input stream or buffer.
+        isSlowMode = true;
```

Comment:
Should `isSlowMode = true` be in `terminate()` so that it is also set on other conditions that result in a terminated state?

Reply:
Done.